src/util/bucket_logs_utils.js

```diff
@@ -67,33 +67,44 @@ async function _upload_to_targets(s3_connection, log_file, bucket_to_owner_keys_
             const target_bucket = log_entry.log_bucket;
             const log_prefix = log_entry.log_prefix;
             const source_bucket = log_entry.source_bucket;
-            if (!bucket_streams[source_bucket + BUCKET_NAME_DEL + target_bucket]) {
+            let upload_stream = bucket_streams[source_bucket + BUCKET_NAME_DEL + target_bucket];
+            if (!upload_stream) {
                 /* new stream is needed for each target bucket, but also for each source bucket
                            - as mulitple buckets can't be written to the same object */
                 const date = new Date();
-                const upload_stream = new stream.PassThrough();
+                upload_stream = new stream.PassThrough();
                 let access_keys;
                 try {
                     access_keys = await bucket_to_owner_keys_func(target_bucket);
                 } catch (err) {
                     dbg.warn('Error when trying to resolve bucket keys', err);
                     if (err.rpc_code === 'NO_SUCH_BUCKET') return; // If the log_bucket doesn't exist any more - nowhere to upload - just skip
                 }
-                s3_connection.config.credentials = new AWS.Credentials(access_keys[0].access_key, access_keys[0].secret_key);
+                //target bucket can be in either cloud or an NC noobaa bucket
+                //use the s3_conn_func CB to get suitable connection
+                const s3_connection = s3_conn_func(access_keys);
                 const sha = crypto.createHash('sha512').update(target_bucket + date.getTime()).digest('hex');
-                promises.push(sem.surround(() => P.retry({
```

Comment:
Removing retries, will it impact the bucket logging?

Reply:
I don't know how well the retry will behave with the v3 Upload.
WCS we will re-process the log file in the next run.